sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIOUtils.java

```diff
@@ -129,19 +131,80 @@ static Map<String, Object> getOffsetConsumerConfig(
     return offsetConsumerConfig;
   }
 
-  // Maintains approximate average over last 1000 elements
-  static class MovingAvg {
+  /*
+   * Attempt to prevent false sharing by padding to at least 64 bytes.
+   * object header: 4, 8, 12 or 16 bytes
+   * alignment: at least 8 bytes
+   */
+  @SuppressFBWarnings("UUF_UNUSED_FIELD")
+  private static class MovingAvgPadding {
+    byte p000, p001, p002, p003, p004, p005, p006, p007;
```

Comment:
Was this a reaction to a measured slowdown? I'm not like super opposed to having something like this, but I expect it's relevance and effectiveness to wane as the codebase evolves, unless we have some fairly deep and clear understanding of exactly what needs to be maintained.

Reply:
Removed layout padding, see comment below.